rero_ils/modules/fees/api.py

```diff
@@ -0,0 +1,115 @@
+from ..circ_policies.api import CircPolicy
+from ..fetchers import id_fetcher
+from ..locations.api import Location
+from ..minters import id_minter
+from ..providers import Provider
+
+# fee provider
+feeProvider = type(
+    'FeeProvider',
+    (Provider,),
+    dict(identifier=FeeIdentifier, pid_type='fee')
+)
+# fee minter
+fee_id_minter = partial(id_minter, provider=feeProvider)
+# fee fetcher
+fee_id_fetcher = partial(id_fetcher, provider=feeProvider)
+
+
+class FeesSearch(RecordsSearch):
+    """RecordsSearch for Fees."""
+
+    class Meta:
+        """Search only on Fees index."""
+
+        index = 'fees'
+
+
+class Fee(IlsRecord):
+    """Fees class."""
+
+    minter = fee_id_minter
+    fetcher = fee_id_fetcher
+    provider = feeProvider
+    model_cls = FeeMetadata
+
+    def create_fee_from_notification(notification):
+        """Create a new fee."""
+        if notification.get('notification_type') == 'overdue':
+            data = {}
+            schemas = current_app.config.get('RECORDS_JSON_SCHEMA')
+            data_schema = {
+                'base_url': current_app.config.get(
+                    'RERO_ILS_APP_BASE_URL'
+                ),
+                'schema_endpoint': current_app.config.get(
+                    'JSONSCHEMAS_ENDPOINT'
+                ),
+                'schema': schemas['fee']
+            }
+            data['$schema'] = '{base_url}{schema_endpoint}{schema}'\
+                .format(**data_schema)
+            base_url = current_app.config.get('RERO_ILS_APP_BASE_URL')
+            url_api = '{base_url}/api/{doc_type}/{pid}'
+            data['creation_date'] = datetime.now(timezone.utc).isoformat()
+            data['fee_type'] = 'overdue'
+            data['notification'] = {
+                '$ref': url_api.format(
+                    base_url=base_url,
+                    doc_type='notifications',
+                    pid=notification.pid)
+            }
+            location_pid = notification.transaction_location_pid
+            data['location'] = {
+                '$ref': url_api.format(
+                    base_url=base_url,
+                    doc_type='locations',
+                    pid=location_pid)
+            }
+            library_pid = Location.get_record_by_pid(location_pid).library_pid
+            patron_type_pid = notification.patron.patron_type_pid
+            holding_circulation_category_pid = notification\
+                .item.holding_circulation_category_pid
+            cipo = CircPolicy.provide_circ_policy(
+                library_pid,
+                patron_type_pid,
+                holding_circulation_category_pid
+            )
+            data['amount'] = cipo.get('overdue_amount')
+            data['currency'] = current_app.config\
+                .get('RERO_ILS_DEFAULT_CURRENCY')
+            data['status'] = 'open'
+            record = Fee.create(
+                data, dbcommit=True, reindex=True, delete_pid=True)
+            return record
```

Comment:
What do we return if we do not have a notification_type == `overdue`?

Reply:
Done